pkg/db/factory.go

```diff
@@ -36,6 +41,28 @@ func WithMigrationTimeout(timeout time.Duration) FactoryOption {
 	}
 }
 
+func WithEncryptionConfiguration(ctx context.Context, configPath string) (FactoryOption, error) {
+	encryptionConf, err := encryptionconfig.LoadEncryptionConfig(ctx, configPath, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Although the config reading code expects GroupResources, we expect GroupKinds,
+	// so just convert, assuming that the Resource is actually a Kind, but lowercase.
+	transformers := make(map[schema.GroupKind]value.Transformer, len(encryptionConf.Transformers))
+	for gr, t := range encryptionConf.Transformers {
+		if len(gr.Resource) < 2 {
```

Comment:
Is there a reason we expect it to be greater than 2?(because the kind is has to be greater than 2 charaters?)

Reply:
This is basically just a sanity check. Since I will be indexing the string, I wanted to avoid weird cases where someone accidentally specifying a 1-character kind, so that the program doesn't just crash